src/app/app.cpp

```diff
@@ -45,9 +45,17 @@ int Application::onExecute() {
         Uint64 start = SDL_GetPerformanceCounter();
 
         while (SDL_PollEvent(&event)) {
-            onEvent(&event);
+            if (event.type == SDL_QUIT) {
+                is_running_m = false;
+            }
+            if (event.type == SDL_KEYDOWN) {
+                if (event.key.keysym.sym == SDLK_ESCAPE) {
+                    is_running_m = false;
+                }
+            }
```

Comment:
Consider removing inner if with 
```suggestion
            if (event.type == SDL_KEYDOWN && event.key.keysym.sym == SDLK_ESCAPE ) {
                    is_running_m = false;
                }
          
```

Reply:
Fixed in 64cf014